### PR TITLE
Added Query Plan Telemetry

### PR DESCRIPTION
**Pull request by laurennat**
Added  a telemetry point for opening query plans

### CODE REVIEW

File: src/queryResult/queryResultWebViewController.ts [hunk 1]
```diff
@@ -327,6 +331,10 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
             } catch (e) {
                 newState.loadState = ApiStatus.Error;
                 newState.errorMessage = e.toString();
+                sendActionEvent(
```
Comment (Benjin) [inline src/queryResult/queryResultWebViewController.ts:334]: `sendErrorEvent` for errors; also add metrics

File: src/queryResult/queryResultWebViewController.ts [hunk 2]
```diff
@@ -305,6 +305,10 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
         state: qr.QueryResultWebviewState,
         xmlPlans: string[],
     ) {
+        sendActionEvent(
```
Comment (Benjin) [inline src/queryResult/queryResultWebViewController.ts:308]: `sendActionEvent`) (and `sendErrorEvent()`) both have optional props that can be set: `additionalMeasurements` (most numeric metrics) and `additionalProperties` (everything else).

Capturing something like the number of plans as a metric here would be good so we can see if, say, people are frequently viewing plans for 25 queries at once, or if we always get errors when it hits some threshold.

File: src/controllers/executionPlanWebviewController.ts [hunk 3]
```diff
@@ -141,6 +146,10 @@ export class ExecutionPlanWebviewController extends ReactWebviewPanelController<
     }
 
     private async createExecutionPlanGraphs(content: string) {
+        sendActionEvent(
+            TelemetryViews.QueryPlan,
+            TelemetryActions.OpenQueryPlan,
+        );
```
Comment (Benjin) [inline src/controllers/executionPlanWebviewController.ts:152]: sending an event at the beginning can be useful, but we generally care more about the end state: did it succeed or fail?  if it failed, what error did we get?  how many execution plans were in the XML?  how long did generating the plan from the XML take?

**Review by Benjin — CHANGES_REQUESTED** [review]

**Comment by laurennat** [comment]
Closing this PR; I'm adding the telemetry changes with my other pr that addresses query plan issues

**Pull request closed without merging**